sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -412,26 +434,27 @@ AZ_NODISCARD az_result setup_response_redirect(CURL * p_curl, az_span * response
 
  * @return AZ_OK if request was sent and a response was received
  */
-AZ_NODISCARD az_result az_http_client_send_request_impl_process(
+static AZ_NODISCARD az_result _az_http_client_curl_send_request_impl_process(
     CURL * p_curl,
     az_http_request * p_request,
     az_http_response * response) {
   az_result result = AZ_ERROR_ARG;
 
-  AZ_RETURN_IF_CURL_FAILED(setup_headers(p_curl, p_request));
+  AZ_RETURN_IF_FAILED(_az_http_client_curl_setup_headers(p_curl, p_request));
 
-  AZ_RETURN_IF_CURL_FAILED(setup_url(p_curl, p_request));
+  AZ_RETURN_IF_FAILED(_az_http_client_curl_setup_url(p_curl, p_request));
 
-  AZ_RETURN_IF_CURL_FAILED(setup_response_redirect(p_curl, &response->_internal.http_response));
+  AZ_RETURN_IF_FAILED(
+      _az_http_client_curl_setup_response_redirect(p_curl, &response->_internal.http_response));
 
   if (az_span_is_equal(p_request->_internal.method, az_http_method_get())) {
-    result = az_curl_send_get_request(p_curl);
+    result = _az_http_client_curl_send_get_request(p_curl);
```

Comment:
This doesn't seem right to me: A get request is very likely to have headers with it - how would they get passed?

Also, I think you don't actually need separate functions for each HTTP method verb. I think you can do this all with the same function and just set the passed-in request method verb.

And finally, Don't access internal fields directly